routing/router.go

```diff
@@ -288,6 +289,12 @@ type Config struct {
 	// we need in order to properly maintain the channel graph.
 	ChainView chainview.FilteredChainView
 
+	// Notifier is a reference to the ChainNotifier, used to grab
+	// the latest blocks if the router is missing any.
+	Notifier chainntnfs.ChainNotifier
```

Comment:
This could be behind an interface in the `routing` package so that `chainntnfs` doesn't have to be imported.  Not necessary for this PR, but this does increase inter-package dependency which could lead to import cycles.

Reply:
+1 on that 

Also perhaps it's possible to instead leverage the `ChainView` which is basically a thin wrapper around an underlying notifier-like abstraction?